generator/src/main/java/com/scottlogic/deg/generator/fieldspecs/FieldSpec.java

```diff
@@ -200,8 +196,37 @@ public FieldSpec withMustContainRestriction(MustContainRestriction mustContainRe
             this.source);
     }
 
-    public FieldSpecSource getFieldSpecSource() {
-        return this.source;
+    /** Create a predicate that returns TRUE for all (and only) values permitted by this FieldSpec */
+    Predicate<Object> getValueValidityPredicate() {
```

Comment:
there's no reason for this to be a predicate

Reply:
This method's gone on a bit of a journey - I agree it's less than compelling in its current form. Would you replace with just e.g. `FieldSpec.permits(Object)`?

Reply:
yeah